txn/flusher.go

```diff
@@ -998,6 +1004,23 @@ func tokensToPull(dqueue []tokenAndId, pull map[bson.ObjectId]*transaction, dont
 	return result
 }
 
+// cleanQueue takes an existing queue and removes all the items in pull from it, generating a new txn-queue.
+func cleanQueue(queue []token, pull map[bson.ObjectId]*transaction, dontPull token) ([]token) {
+	cleaned := make([]token, 0, len(queue))
```

Comment:
When filtering like this, a nice trick I like is:
`cleaned := queue[:0]`
Because the slice is created with the same underlying array, there's no re-allocation up to the original length.